app/services/course/assessment/submission/statistics_download_service.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+require 'csv'
+class Course::Assessment::Submission::StatisticsDownloadService
+  class << self
+    # Downloads the submissions statistics and zip it.
+    #
+    # @param [CourseUser] course_user The course user downloading the statistics.
+    # @param [Course::Assessment] assessment The assessments to download statistics from.
+    # @param [String|nil] students The subset of students whose submissions' statistics to download.
+    # Accepted values: 'my', 'phantom'
+    # @return [String] The path to the zip file.
+    def download_and_zip(course_user, assessment, students)
```

Comment:
Can I suggest we handle the params 'my' and 'phantom' at the controller? 

That would simplify this service to just accept 1 parameter, which are the collection of submissions to be zipped and downloaded.

Reply:
OK, no problem, will change that.